examples/How_to_use_guardrails.ipynb

```diff
@@ -0,0 +1,462 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 57,
+   "id": "e95efc89",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "system_prompt = \"You are a helpful assistant.\"\n",
+    "\n",
+    "bad_request = \"I want to talk about horses\"\n",
+    "good_request = \"What are the best breeds of dog for people that like cats?\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 69,
+   "id": "fee948e2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import asyncio\n",
+    "\n",
+    "\n",
+    "async def get_chat_response(user_request):\n",
+    "    print(\"Getting LLM response\")\n",
+    "    messages = [\n",
+    "        {\"role\": \"system\", \"content\": system_prompt},\n",
+    "        {\"role\": \"user\", \"content\": user_request},\n",
+    "    ]\n",
+    "    response = openai.ChatCompletion.create(\n",
+    "        model=GPT_MODEL, messages=messages, temperature=0.5\n",
+    "    )\n",
+    "    print(\"Got LLM response\")\n",
+    "    return response[\"choices\"][0][\"message\"][\"content\"]\n",
+    "\n",
+    "\n",
+    "async def topical_guardrail(user_request):\n",
+    "    print(\"Checking topical guardrail\")\n",
+    "    messages = [\n",
+    "        {\n",
+    "            \"role\": \"system\",\n",
+    "            \"content\": \"Your role is to assess whether the user question is allowed or not. The allowed topics are cats and dogs. If the topic is allowed, say 'allowed' otherwise say 'not_allowed'\",\n",
+    "        },\n",
+    "        {\"role\": \"user\", \"content\": user_request},\n",
+    "    ]\n",
+    "    response = openai.ChatCompletion.create(\n",
+    "        model=GPT_MODEL, messages=messages, temperature=0\n",
+    "    )\n",
+    "    print(\"Got guardrail response\")\n",
+    "    return response[\"choices\"][0][\"message\"][\"content\"]\n",
+    "\n",
+    "\n",
+    "async def execute_chat_with_guardrail(user_request):\n",
+    "    topical_guardrail_task = asyncio.create_task(topical_guardrail(user_request))\n",
+    "    chat_task = asyncio.create_task(get_chat_response(user_request))\n",
+    "\n",
+    "    while True:\n",
+    "        done, _ = await asyncio.wait(\n",
+    "            [topical_guardrail_task, chat_task], return_when=asyncio.FIRST_COMPLETED\n",
+    "        )\n",
+    "        for task in done:\n",
+    "            if task == topical_guardrail_task:\n",
+    "                guardrail_response = topical_guardrail_task.result()\n",
+    "                if guardrail_response == \"not_allowed\":\n",
+    "                    chat_task.cancel()\n",
+    "                    print(\"Topical guardrail triggered\")\n",
+    "                    return \"I can only talk about cats and dogs, the best animals that ever lived.\"\n",
+    "            elif task == chat_task:\n",
+    "                chat_response = chat_task.result()\n",
+    "                if (\n",
+    "                    not topical_guardrail_task.done()\n",
```

Comment:
If the guardrail task is not done, we would not want to return the chat_response yet, right?

Reply:
Good catch, yes - updated now